### DISCUSSION TITLE
How do you type your `$model->getKey()` ?

### DISCUSSION THREAD

**Discussion by treyssatvincent** [larastan/larastan#2085] [General]
This handy method on models to get the key value is typed as `mixed` by the framework, it's not ideal because when using it on other methods it's easy to get "Unable to resolve the template type ..." errors.

I was wondering how do you solve this on your projects ?

**Comment by calebdw** [comment]
This just needs to be fixed in the framework, I'm pretty certain model keys can only be `int|string` (or `array-key` if you're feeling fancy)

**Reply by treyssatvincent** [reply]
Sure, but even with that there's case when you want to know if it's a `string` or a `int`.

**Reply by calebdw** [reply]
We know all the modal database properties and their types, it should be possible to create a dynamic return type extension that returns the correct type.

However, most of the time when I find myself using the `getType` method is when I only have a `Model` instance and don't know the specific model so not sure how useful this will really be

**Comment by canvural** [comment]
`array-key` would work because it's a benevolent union type. So it'll not complain if a method expects an `int` argument and you pass `$model->getKey()`

It's a good idea to PR it to the framework 👍🏽 But I think we can also make the change here with a stub file and that way we can release faster and more than the latest version would have it. So feel free to open a PR

**Comment by treyssatvincent** [comment]
I opened [a pull request on the framework](https://github.com/laravel/framework/pull/53396), but note that it can also be null (on pivots) : as specified in [EloquentPivotTest](https://github.com/laravel/framework/blob/cdf07b6678ea64a2a5a9fbe83df9bf4a6586c0a7/tests/Integration/Database/EloquentPivotTest.php#L63).
